src/main/java/hillel/doctorRest/clinic/schedule/ScheduleService.java

```diff
@@ -5,33 +5,61 @@
 
 import java.time.LocalDate;
 import java.util.List;
-import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.val;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @AllArgsConstructor
-
 public class ScheduleService {
     private final ScheduleRepository scheduleRepository;
 
-    public final List<Schedule> findAll() {
+    public Schedule createSchedule(Integer doctorId,
+                                   LocalDate visitDate,
+                                   String hour, Schedule schedule) {
+        schedule.setDoctorId(doctorId);
+        schedule.setVisitDate(visitDate);
+        schedule.setHour(hour);
+        return saveSchedule(schedule);
+    }
+
+    public List<Schedule> findAll() {
         return scheduleRepository.findAll();
     }
 
+    public List<Schedule> findByDoctorId(Integer id) {
+        return scheduleRepository.findByDoctorId(id);
+    }
+
     public List<Schedule> findByDoctorIdAndVisitDate(Integer id,
                                                      LocalDate visitDate) {
         return scheduleRepository.findByDoctorIdAndVisitDate(id, visitDate);
     }
 
-    public Optional<Schedule> findByHour(String hour) {
-        return scheduleRepository.findByHour(hour);
+    public List<Schedule> findByDoctorIdAndVisitDateAndHour(Integer doctorId,
+                                                            LocalDate visitDate,
+                                                            String hour) {
+        return scheduleRepository.findByDoctorIdAndVisitDateAndHour(doctorId, visitDate, hour);
     }
 
-    public Schedule createSchedule(Integer doctorId,
-                                   LocalDate visitDate,
-                                   String hour, Schedule schedule) {
-        schedule.setDoctorId(doctorId);
-        schedule.setVisitDate(visitDate);
-        schedule.setHour(hour);
+    @Transactional(rollbackFor = Exception.class)
+    public void swapListDoctors(LocalDate visitDate, Integer doctor1Id, Integer doctor2Id) throws Exception {
```

Comment:
It is bad practice to use checked exceptions. So please try to avoid using that 'throws'
Also, `@Transactional` works as expected on unchecked exceptions, so you can remove `rollbackFor` too.